feat(django-homework-assignment3): Add inventory changelog.

morristseng opened the pull request:
[why]
  Foe the django homework assignment3
    
[how]
  In order to get who change the record. I created an admin model for
  inventory to retrive "who" from the request object.
    
  Same as the changed fields. Get this info from "form.changed_data"

morristseng committed:
feat(django-homework-assignment3): Add inventory changelog.

[why]
Foe the django homework assignment3

[how]
In order to get who change the record. I created an admin model for
inventory to retrive "who" from the request object.

Same as the changed fields. Get this info from "form.changed_data"

morristseng committed:
refactor(django-homework-assignment3): Use template to reduce copy-paste codes.

[why]
The html files had too many duplicate cades. Need to remove it.

[how]
Use template engine to extract "api_route" and "columns" attributes.
Thus, we can combine all html files into one file.

alfredweihc commented on inventory/signals.py:
```diff
@@ -0,0 +1,29 @@
+    if action_type == 'add':
+        details = f"{obj.id} is created"
+    elif action_type == 'modify':
+        details = f"{obj.id}'s field: {','.join(updated)} {'are' if len(updated) > 1 else 'is'} changed"
```
nit: detail 如果是文字描述 255 可能太短

morristseng replied:
Enlarge to 2048.

alfredweihc commented on inventory/templates/inventory/list.html:
```diff
@@ -22,7 +22,7 @@
 
     React.useEffect(() => {
-      fetch(`/inventory/api/inventory/status?o=${rows}&page=${page + 1}`)
+      fetch(`/inventory/api/inventory/{{ api_route }}?o=${rows}&page=${page + 1}`)
```
api 前面多一個 inventory 好像有點多餘

morristseng replied:
Removed.

alfredweihc reviewed (approved):
overall lgtm

morristseng committed:
refactor(django-homework-assignment3): Enlarge length of log detail and remove "inventory" prefix for api endpoint.

[why]
Address reviewer comments

[how]
Enlarge max length of log detail to 2048.
Move api enpoints to api_urls.py in order to remove "inventory" prefix
for all api route.
